tctl: users add/ls and tokens ls json output

klizhentas opened the pull request:
This commit adds --format flag (that is already present in other subcommands) support for `tctl users ls` and `tctl tokens ls`

Commit:
tctl: users add/ls and tokens ls json output

klizhentas commented:
retest this please

r0mant commented on lib/services/invite.go:
```diff
@@ -0,0 +1,63 @@
+
+// InviteTokenV3 is an invite token spec format V3
+type InviteTokenV3 struct {
+	// Kind is a resource kind - always resource.
```
always "invite" maybe?

r0mant commented on tool/tctl/common/token_command.go:
```diff
@@ -178,18 +183,26 @@ func (c *TokenCommand) List(client auth.ClientI) error {
+			return table.AsBuffer().String()
+		}
+		fmt.Printf(tokensView())
+	} else {
```
Is "c.format" validated anywhere that it is either "text" or "json"? If not, this will accept anything (--format=asd) and render json instead of producing error. Same q for similar blocks below.

klizhentas committed:
Allow RemoteProxy to read local Roles

RemoteProxy is allowed to read users, but not roles,
what breaks streaming caching that is set up to fetch users
and roles.

This commit allows remote proxy to read roles and users.

russjones commented on lib/services/invite.go:
```diff
@@ -0,0 +1,63 @@
+)
+
+// InviteTokenV3 is an invite token spec format V3
+type InviteTokenV3 struct {
```
Shouldn't this be defined in `types.proto`?

russjones commented on tool/tctl/common/user_command.go:
```diff
@@ -63,6 +67,7 @@ func (u *UserCommand) Initialize(app *kingpin.Application, config *service.Confi
 	u.userAdd.Flag("ttl", fmt.Sprintf("Set expiration time for token, default is %v hour, maximum is %v hours",
 		int(defaults.SignupTokenTTL/time.Hour), int(defaults.MaxSignupTokenTTL/time.Hour))).
 		Default(fmt.Sprintf("%v", defaults.SignupTokenTTL)).DurationVar(&u.ttl)
+	u.userAdd.Flag("format", "Output format, 'text' or 'json'").Hidden().Default(teleport.Text).StringVar(&u.format)
```
Is this hidden because it's not documented yet?

Pluggi commented:
Has this been dropped? Do you plan to add it to 4.0.0?
